cmsdev/internal/test/ims/verify_ims_images.go

```diff
@@ -34,32 +35,58 @@ import (
  *
  */
 
-func testPackage() {
-	common.Infof("Testing package")
+// Test image CRUD operations using all supported API versions
+func TestImageCRUDOperationUsingAPIVersions() (passed bool) {
+	passed = true
+
+	for _, version := range common.IMSAPIVERSIONS {
+		common.PrintLog(fmt.Sprintf("Testing image CRUD operations using API version: %s", version))
+		common.SetIMSAPIVersion(version)
+		passed = passed && TestImageCRUDOperation(version)
+	}
+
+	// default API version
+	common.SetIMSAPIVersion("")
+	common.PrintLog("Testing image CRUD operations using default API version")
+	passed = passed && TestImageCRUDOperation(common.GetIMSAPIVersion())
+	return passed
 }
 
-func TestImageCRUDOperation() (passed bool) {
+func TestImageCRUDOperation(apiVersion string) (passed bool) {
 	// Test creating an image
 	imageRecord, success := TestImageCreate()
 	if !success {
 		return false
 	}
-	// Test updating the image
-	updated := TestImageUpdate(imageRecord.Id)
 
-	// Test soft deleting the image
-	deleted := TestImageDelete(imageRecord.Id)
+	if apiVersion == "v3" {
+		// Test updating the image
+		updated := TestImageUpdate(imageRecord.Id)
```

Comment:
Since this is the beginning of both branches of the conditional, wouldn't it make sense to have it before the conditional?